src/runtime_src/core/common/api/bo_int.h

```diff
@@ -27,6 +27,11 @@ XRT_CORE_COMMON_EXPORT
 xrt::bo
 create_debug_bo(const xrt::hw_context& hwctx, size_t sz);
 
+// dump_bo() - Dump bo contents to a file
+XRT_CORE_COMMON_EXPORT
```

Comment:
Compiler teams (xcompiler, aie-compiler) have their our runner. This function is for sure make their life easier.

Reply:
If they are like external clients of XRT, then they cannot access this function after making it internal because  no externally exported header file declares it.  And I still stand by it being an internal function and not exposing in public API.

Reply:
Oh. I see function
read() - Copy-out user data from host backing storage of BO.

That is what compiler team is using.